v1.0.3

boxdrop opened the pull request:
Added support for adding insurance to shipments, change carrier, plus bugfixes

Commit:
#921 file creation umask and chmod

Commit:
#832	admin folder guessing for setups without "admin" in admin folder name

Commit:
#923 missing translation for save button

Commit:
#861 add insurance option

Commit:
#861 translation for insurance

Commit:
#861 Validation corrections for upgrade script

Commit:
#645 change carrier for non-boxdrop orders

Quetzacoalt91 commented on lib/BoxdropHelper.class.php:
```diff
@@ -257,4 +263,43 @@ public static function explodeString($delimiter, $input)
+	        $return = str_replace(',', '.', $return);
+	
+	      return (float) $return;
+	    }
```
You should also try this function to check if it fits your needs:

``` php
    public static function convertFormNumberToFloat($number)
    {
        return (float)str_replace(array('.', ','), array('','.'), $return);
    }
```

Quetzacoalt91 commented on boxdropshipment.php:
```diff
@@ -143,7 +143,8 @@ public function install()
 
-			if (!$this->registerHook('actionValidateOrder') || !$this->registerHook('displayCarrierList') || !$this->registerHook('updateCarrier'))
+			if (!$this->registerHook('actionValidateOrder') || !$this->registerHook('displayCarrierList') ||
+			!$this->registerHook('updateCarrier') || !$this->registerHook('displayAdminOrderContentShip'))
```
Adding the new hook registration would be good in the upgrade file too.

Quetzacoalt91 commented on upgrades/Upgrade-1.0.2.php:
```diff
@@ -0,0 +1,35 @@
+{
+	return Db::getInstance()->execute('ALTER TABLE `'._DB_PREFIX_.'boxdrop_order_shipment` ADD
+	 `insurance_amount` DECIMAL(7,2) NOT NULL DEFAULT \'0.00\' 
+	 AFTER `airwaybill`;');
```
You did not update the CREATE TABLE request in _sql/create.php_, so this field will be missing on a fresh install.

Quetzacoalt91 commented on lib/BoxdropHelper.class.php:
```diff
@@ -45,10 +47,15 @@ public static function getAdminDirPath()
+							{								
+								$admin_dir = $absolute_element;
+								break;
+							}
```
I guess we could improve the way you create the admin environment for the next version of BoxDrop, in example with ModuleAdminController (and a ModuleFrontController for the front-office). When the controller will be called, you will be sure to be in an admin context, and all your checks won't be needed anymore. Please have a look on it, I'm sure your code will be safer and cleaner.

Commit:
- Changes related to PR comments
- Changed hook displayAdminOrderContentShip to displayAdminOrder in order to support older versins of PrestaShop

boxdrop commented:
Thanks for you feedback, we have pushed the changes now.

Commit:
Rename update file

boxdrop commented:
comment.

Quetzacoalt91 commented:
Wouuups ! I have just seen that your upgrade folder is wrong ! It should be *upgrade*, not *upgrades*.

Commit:
renamed upgrades folder to upgrade

boxdrop commented:
There you go!

Quetzacoalt91 commented:
Thanks @boxdrop, the pull-request has just been merged and the changes will be available from tomorrow.

Regards
